File: placementprepration/towerofhanoi.java
import java.util.*;
public class towerofhanoi{
    public static void toh(int n, char src, char aux, char dest){
        if(n==1){
            System.out.println("Move disk 1 from " + src + " to " + dest);
            return ;
        }
        toh(n-1,src,dest,aux);
        System.out.println("Move disk " + n + " from " + src + " to " + dest);
        toh(n-1,aux,src,dest);
    }
    public static void main(String[] args){
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();
        toh(n,'a','b','c');
    }

}

File: placementprepration/towerofhanoi.py
import sys


def toh(n, src, aux, dest):
    if n == 1:
        print(f"Move disk 1 from {src} to {dest}")
        return
    toh(n - 1, src, dest, aux)
    print(f"Move disk {n} from {src} to {dest}")
    toh(n - 1, aux, src, dest)


def main():
    data = sys.stdin.read().strip().split()
    if not data:
        return
    n = int(data[0])
    toh(n, 'a', 'b', 'c')


if __name__ == "__main__":
    main()
